assets/blocks/reader-registration/edit.js

```diff
@@ -12,30 +21,197 @@ import './editor.scss';
+											}
+										} }
+									/>
+								) ) }
+								<p>
+									<a href={ newspack_blocks.newsletters_url }>
+										{ __( 'Manage your subscription lists', 'newspack-newsletters' ) }
+									</a>
+								</p>
+							</>
+						) }
+					</PanelBody>
+				) }
 			</InspectorControls>
 			<div { ...blockProps }>
-				<div className="newspack-registration">
+				<div className={ `newspack-registration ${ className }` }>
 					<form onSubmit={ ev => ev.preventDefault() }>
-						<input type="email" placeholder={ placeholder } />
-						<input type="submit" value={ label } />
+						<div className="newspack-registration__form-content">
+							{ newsletterSubscription && lists.length ? (
+								<div className="newspack-registration__lists">
+									{ lists?.length > 1 && (
+										<RichText
+											onChange={ value => setAttributes( { newsletterTitle: value } ) }
+											placeholder={ __( 'Newsletters title…', 'newspack' ) }
+											value={ newsletterTitle }
+											tagName="h3"
+										/>
+									) }
+									<ul>
+										{ lists.map( listId => (
+											<li key={ listId }>
+												<span className="newspack-registration__lists__checkbox">
+													<input type="checkbox" checked readOnly />
+												</span>
+												<span className="newspack-registration__lists__details">
+													<span className="newspack-registration__lists__label">
+														<span className="newspack-registration__lists__title">
+															{ lists.length === 1 ? (
+																<RichText
+																	onChange={ value => setAttributes( { newsletterLabel: value } ) }
+																	placeholder={ __( 'Subscribe to our newsletter', 'newspack' ) }
+																	value={ newsletterLabel }
+																	tagName="span"
+																/>
+															) : (
+																listConfig[ listId ]?.title
+															) }
+														</span>
+														{ displayListDescription && (
+															<span className="newspack-registration__lists__description">
+																{ listConfig[ listId ]?.description }
+															</span>
+														) }
+													</span>
+												</span>
+											</li>
+										) ) }
+									</ul>
+								</div>
+							) : null }
+							<div className="newspack-registration__main">
+								<div className="newspack-registration__inputs">
+									<input type="email" placeholder={ placeholder } />
+									<button type="submit">
+										<RichText
+											onChange={ value => setAttributes( { label: value } ) }
+											placeholder={ __( 'Register', 'newspack' ) }
+											value={ label }
+											tagName="span"
+										/>
+									</button>
+								</div>
+								<div className="newspack-registration__privacy">
+									<RichText
+										onChange={ value => setAttributes( { privacyLabel: value } ) }
+										placeholder={ __( 'Newsletters title…', 'newspack' ) }
```

Comment:
This placeholder text should more accurately reflect the purpose:

```suggestion
										placeholder={ __( 'Terms & Conditions statement…', 'newspack' ) }
```